tools/cldr/cldr-to-icu/src/main/java/org/unicode/icu/tool/cldrtoicu/IcuTextWriter.java

```diff
@@ -0,0 +1,311 @@
+                    }
+                }
+            } else {
+                // Only one array for the rbPath, so don't add an extra set of braces.
+                printArray(onlyValue, quote, isSequence, out);
+            }
+        } else {
+            for (RbValue value : values) {
+                if (value.size() == 1) {
+                    // Single-value array: print normally.
+                    printArray(value, quote, isSequence, out);
+                } else {
+                    // Enclose this array in braces to separate it from other values.
+                    open("", out);
+                    printArray(value, quote, isSequence, out);
+                    close(out);
+                }
+            }
+        }
+        return wasSingular;
+    }
+
+    private static final RbPath RB_SEQUENCE = RbPath.of("Sequence");
+    private static final RbPath RB_RULES = RbPath.of("rules");
+    private static final RbPath RB_LOCALE_SCRIPT = RbPath.of("LocaleScript");
+    private static final RbPath RB_ERAS = RbPath.of("eras");
+    private static final RbPath RB_NAMED = RbPath.of("named");
+    private static final RbPath RB_CALENDAR_PREFERENCE_DATA = RbPath.of("calendarPreferenceData");
+    private static final RbPath RB_METAZONE_INFO = RbPath.of("metazoneInfo");
+
+    /**
+     * Wrapper for a hack to determine if the given rb path should always present its values as an
+     * array.
+     */
+    // TODO: Verify this is still needed, and either make it less hacky, or delete it.
+    private static boolean mustBeArray(boolean topValues, String name, RbPath rbPath) {
+        if (topValues) {
+            // matches "rules/setNN" (hence the mucking about with raw segments).
+            return name.equals("pluralRanges")
+                && rbPath.startsWith(RB_RULES)
+                && rbPath.getSegment(1).startsWith("set");
+        }
+        return rbPath.equals(RB_LOCALE_SCRIPT)
+            || (rbPath.contains(RB_ERAS)
+                && !rbPath.getSegment(rbPath.length() - 1).endsWith(":alias")
+                && !rbPath.endsWith(RB_NAMED))
+            || rbPath.startsWith(RB_CALENDAR_PREFERENCE_DATA)
+            || rbPath.startsWith(RB_METAZONE_INFO);
+    }
+
+    private void printArray(RbValue rbValue, boolean quote, boolean isSequence, PrintWriter out) {
+        for (int n = 0; n < rbValue.size(); n++) {
+            newLineAndIndent(out);
+            printValue(out, quoteInside(rbValue.getElement(n)), quote);
+            if (!isSequence) {
+                out.print(",");
+            }
+        }
+    }
+
+    private static void printValue(PrintWriter out, String value, boolean quote) {
+        if (quote) {
+            out.append('"').append(value).append('"');
+        } else {
+            out.append(value);
+        }
+    }
+
+    // Can a string be broken here? If not, backup until we can.
+    private static int goodBreak(String quoted, int end) {
```

Comment:
should this actually use the line break iterator (that's what it's for!!) to backup to a base char? i.e. to not split `é` into `e`+\n+`´`? A previous ICU's LB iterator should be fine.

Reply:
Ahh, thanks. I'm actually not super familiar with this, so I'll see about making a change, but you'll have to check it carefully to make sure I'm doing the right thing (I'll add unit tests since this is going to be rarely hit IRL).

Reply:
Mind if I do this in a followup CL (and add a TODO now) since it seems more than just a few lines changed? Or, we could just not line wrap at all since it doesn't seem structurally necessary (though this should be done later since it changes output) ??

Reply:
Yes, looks more complicated. It should use a linebreak iterator, but it should also have a specialized CharSequence that knows about the escapes that could be used.

Alternatively, if there are any escapes, then unescape the whole string, find the linebreak, then escape (that's likely to be much simpler).

Reply:
Ahh thanks. Seems like since it's producing identical output now, this is probably best handled via a followup ticket/pr. I'll be sure to promote the TODO that I just added to a ticket when I do the sweep.